File: backend/src/test/java/kr/co/programmers/collabond/util/WithCustomMockUser.java
package kr.co.programmers.collabond.util;

import kr.co.programmers.collabond.api.user.domain.Role;
import org.springframework.security.test.context.support.WithSecurityContext;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(factory = WithCustomMockUserSecurityContextFactory.class)
public @interface WithCustomMockUser {
    String username() default "providerId";
    Role role() default Role.ROLE_IP;
}

File: backend/src/test/java/kr/co/programmers/collabond/util/WithCustomMockUser.kt
package kr.co.programmers.collabond.util

import kr.co.programmers.collabond.api.user.domain.Role
import org.springframework.security.test.context.support.WithSecurityContext

@Retention(AnnotationRetention.RUNTIME)
@WithSecurityContext(factory = WithCustomMockUserSecurityContextFactory::class)
annotation class WithCustomMockUser(
    val username: String = "providerId",
    val role: Role = Role.ROLE_IP
)